src/pages/index.js

```diff
@@ -41,38 +49,81 @@ function handleImageClick(name, link) {
 }
 
 //Popup Forms
-const profilePopup = new PopupWithForm(profileModal, (data) => {
-  console.log(data);
-  userInfo.setUserInfo(data);
-  profilePopup.closePopup();
-});
-const cardForm = new PopupWithForm(newCardModal, (data) => {
-  const name = newCardTitle.value;
-  const link = newCardURL.value;
-  addCard({ name, link }, mediaList);
-  cardForm.closePopup();
+const profilePopup = new PopupWithForm(profileModal, (data, button) => {
+  api
+    .saveUserInfo(data)
+    .then(() => {
+      userInfo.setUserInfo(data);
+      profilePopup.closePopup();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally((res) => {
+      button.textContent = "Save";
+    });
+  button.textContent = "Saving...";
 });
+
+const profilePicturePopup = new PopupWithForm(
+  profilePictureModal,
+  (data, button) => {
+    api
+      .updatePicture(data.url)
+      .then((res) => {
+        userInfo.setAvatar = res.avatar;
+        profilePicturePopup.closePopup();
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally((res) => {
+        button.textContent = "Save";
+      });
+    button.textContent = "Saving...";
+  }
+);
+const deletePopup = new PopupWithConfirmation(deleteModal);
+deletePopup.setEventListeners();
+
+function handleDeleteCard(cardId, cardElement) {
+  const id = cardId;
+  deletePopup.openPopup();
+  deletePopup.setSubmitAction(() => {
+    api
+      .deleteCard(id)
+      .then((res) => {
+        cardElement.remove();
+        deletePopup.closePopup();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  });
+}
+// function cancelDeleteCard();
+function handleCardLike(isLiked, cardId) {
+  api
+    .cardLikeToggle(isLiked, cardId)
+    .then((res) => {
+      console.log(res);
+      this.toggleLike();
```

Comment:
`this` should **not** be used outside classes in functions in `index.js` . This context is easy to lose. You need to find other ways of passing data.

Reply:
**Could be improved**

To have access to any method of the class in `index.js` you can pass `this` to the call:

```js

        this._likeButton.addEventListener('click', () => {
            this._handleLike(this);
        });


```

And now in `index.js` you can call the methods like this:


```js


function addLike(cardInstance) {
        api.likeCard(cardInstance.getId())
            .then((res) => {
               cardInstance.addLike(res)
            })
            .catch((err) => {
                console.error(err);
            });
}


```

The same thing is with deleting cards